functions/data_transformations.R

```diff
@@ -0,0 +1,64 @@
+install.packages("jsonlite")
+setwd("/Users/jameshaugen/Documents/git_lesson/iterations-jamhau")
+library(httr)
+library(jsonlite)
+library(ggplot2)
+library(DescTools)
+library(tidyverse)
+library(magrittr)
+library(rlang)
+library(lubridate)
+library(anytime)
+library(readr)
+library(yaml)
+
+#### 1: Beginning of script
+
+# Load function for posting GQL-queries and retrieving data: 
+source("functions/GQL_function.r")
+
+# The URL we will use is stored below: 
+
+configs <- 
+  read_yaml("vegvesen_configs.yml")
+
+
+gql_metadata_qry <- read_file("gql-queries/station_metadata.gql")
+
+# Let's try submitting the query: 
+
+stations_metadata <-
+  GQL(
+    query=gql_metadata_qry,
+    .url = configs$vegvesen_url
+  ) 
+
+# data_transformations.r
+
+
+
+
+transform_metadata_to_df <- function(stations_metadata) {
+  # Extract relevant data from the metadata
+  stations_metadata <- stations_metadata[[1]] |> 
+    map(as_tibble) |> 
+    list_rbind() |> 
+    mutate(latestData = map_chr(latestData, 1, .default = ""))  |> 
+    mutate(latestData = as_datetime(latestData, tz = "UTC"))  |> 
+    mutate(location = map(location, unlist)) |>  
+    mutate(
+      lat = map_dbl(location, "latLon.lat"),
+      lon = map_dbl(location, "latLon.lon")
+    ) %>% 
+    select(-location)
+  
+  return(stations_metadata)
+}
```

Comment:
This looks correct. You do however miss the `transform_volumes` function and the `to_iso8601` function should have been in this file.